src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -148,6 +148,12 @@ internal sealed partial class ConsoleHost
 
             try
             {
+                // We need to read the settings file before we create the console host
+                string[] tempArgs = new string[args.GetLength(0)];
+                args.CopyTo(tempArgs, 0);
```

Comment:
Could you please clarify why we need clone `args`?

Reply:
The existing code cloned the args.  I don't want to break something by not cloning.

Reply:
Reviewing the code.  I don't think it has to be.  I especially don't think it has to be twice.  compromise, I'll remove the second clone, you file an issue to remove the first one.

Reply:
Done #7457.